src/main/java/org/elasticsearch/search/warmer/IndexWarmersMetaData.java

```diff
@@ -74,6 +91,30 @@ public BytesReference source() {
         public Boolean queryCache() {
             return this.queryCache;
         }
+
+        @Override
+        public boolean equals(Object o) {
+            if (this == o) return true;
+            if (o == null || getClass() != o.getClass()) return false;
+
+            Entry entry = (Entry) o;
+
+            if (!name.equals(entry.name)) return false;
+            // Probably incorrect - comparing Object[] arrays with Arrays.equals
```

Comment:
This is automatically generated code by IntelliJ, which also automatically generated this comment. I am not really sure what's wrong with this approach. If you don't see any issues with it - I will remove the comment.